include/seqan3/core/type_list.hpp

```diff
@@ -31,10 +35,485 @@ using type_list = meta::list<types...>;
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <seqan3::detail::TypeList list1_t, seqan3::detail::TypeList list2_t>
+using concat = decltype(detail::concat_impl(list1_t{}, list2_t{}));
+
+/*!\brief Return a seqan3::type_list of all the types in the type list, except the first.
+ * \tparam list_t The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Constant.
+ */
+template <seqan3::detail::TypeList list_t>
+//!\cond
+    requires (size<list_t> > 0)
+//!\endcond
+using drop_front = decltype(detail::drop_front_impl(list_t{}));
+
+/*!\brief Return a seqan3::type_list of the first `n` types in the input type list.
+ * \tparam n        The target size; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using take = typename decltype(detail::split_after_impl<n>(type_list<>{}, list_t{}))::first_type;
+
+/*!\brief Return a seqan3::type_list of the types in the input type list, except the first `n`.
+ * \tparam n        The amount to drop; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using drop = typename decltype(detail::split_after_impl<n>(type_list<>{}, list_t{}))::second_type;
+
+/*!\brief Return a seqan3::type_list of the last `n` types in the input type list.
+ * \tparam n        The target size; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `size<list_t> - n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using take_last = typename decltype(detail::split_after_impl<size<list_t> - n>(type_list<>{}, list_t{}))::second_type;
+
+/*!\brief Return a seqan3::type_list of the types the input type list, except the last `n`.
+ * \tparam n        The amount to drop; must be >= 0 and <= the size of the input type list.
+ * \tparam list_t   The (input) type list.
+ * \ingroup type_traits
+ *
+ * \details
+ *
+ * ### (Compile-time) Complexity
+ *
+ * Equal to `size<list_t> - n` (linear).
+ */
+template <ptrdiff_t n, seqan3::detail::TypeList list_t>
+//!\cond
+    requires (n >= 0 && n <= size<list_t>)
+//!\endcond
+using drop_last = typename decltype(detail::split_after_impl<size<list_t> - n>(type_list<>{}, list_t{}))::first_type;
```

Comment:
accordingly we can model this with the negative index and with take as well. I think it is more readable.

Reply:
I had it like that before, but it's horrible :ghost: 

The problem is that it gets very confusing regarding inclusive/exclusive indexing.

take and drop are specifically not modeled as "take until third element", but as "take three elements". What would `take<-3>` imply? "Take until the third-last"? Before or after? What does it imply regarding the size of the list?

This way it is very simple: `take<n>` and `take_last <n>` always return something of size n. `drop<n>` and `drop_last <n>`  always return something of the original size - n.

range-v3 is also getting a `view::take_last` so we would be in-sync regarding naming.

Reply:
Well, the negative index would perfectly align with take because take<-3> takes exactly the last three elements. That's how it is implemented anyway (`take_last<n> <=> drop<N-n>`). Thus you would have `take<N> == take<-N>` and `take<1> <=> at<0>` and `take<-1> <=> at<N-1> <=> at<-1>`. 
But if the ranges library get's a special take_last then why not having it here as well. But honestly, I don't like that  `at` has then a different semantics. But oh boy,  `at_last` would be the wrong way to go as well 🙈 .

Reply:
> Well, the negative index would perfectly align with take because take<-3> takes exactly the last three elements.

Oh, wow, ok, you thought it would work that way.

> But if the ranges library get's a special take_last then why not having it here as well. 

Yes, I really think it makes it clearer as we are not working with indexes at all, but with counts. And negative counts are weird.

Reply:
> Oh, wow, ok, you thought it would work that way.

It should work that way if `at` works that way!